mlrun/datastore/base.py

```diff
@@ -36,6 +39,141 @@
+
+    # Pattern to match PyArrow schema conflict errors
+    conflict_pattern = (
+        r"Unable to merge: Field (\w+) has incompatible types: (.+?) vs (.+)$"
+    )
+
+    match = re.search(conflict_pattern, error_message.strip())
+    if not match:
+        return None
+
+    field_name = match[1]
+    type1 = match[2].strip()
+    type2 = match[3].strip()
+
+    # Check if this looks like a partition vs data column conflict
+    # Partition columns are typically string type
+    is_partition_conflict = (
+        (type1 == "string" and "dictionary" in type2)
+        or (type2 == "string" and "dictionary" in type1)
+        or (type1 == "string" and type2 != "string")
+        or (type2 == "string" and type1 != "string")
+    )
+
+    if not is_partition_conflict:
+        return None
+
+    # Determine which is partition vs data type
+    if type1 == "string":
+        partition_type = type1
+        data_type = type2
+    else:
+        partition_type = type2
+        data_type = type1
+
+    return {
+        "field_name": field_name,
+        "partition_type": partition_type,
+        "data_type": data_type,
+        "pyarrow_version": str(_get_pyarrow_version()),
+    }
+
+
+def _create_schema_conflict_message(conflict_info):
+    """Create a detailed error message explaining the schema conflict."""
+    field_name = conflict_info["field_name"]
+    partition_type = conflict_info["partition_type"]
+    data_type = conflict_info["data_type"]
+    pyarrow_version = conflict_info["pyarrow_version"]
+
+    message = f"""
+PYARROW SCHEMA CONFLICT DETECTED
+
+Problem:
+The parquet file has a schema conflict for field '{field_name}':
+- Partition column type: {partition_type}
+- Data column type: {data_type}
+
+What changed in PyArrow 17+:
+PyArrow {pyarrow_version} introduced stricter schema validation that prevents automatic
+ type coercion between incompatible types. Previously, PyArrow would silently convert
+ types, which could cause data corruption.
+
+Why this happens:
+Your file path contains partition information (e.g., '{field_name}=value/') which
+creates a partition column with type '{partition_type}'. However, the actual parquet
+ data also contains a '{field_name}' column with type '{data_type}'. PyArrow 17+
+cannot automatically merge these incompatible types.
+
+Solutions:
+
+1. Standardize column types:
+   Ensure partition columns and data columns use the same data types when recreating
+   your dataset. For example, if your data column uses dictionary encoding, ensure
+   the partition column also uses the same type, or convert both to a common type
+   like string.
+
+2. Rename partition columns:
+   Modify your data pipeline to use different names for partition columns to avoid
+   name collisions. For example:
+   - Instead of: date=2023-02-27/file.parquet
+   - Use: partition_date=2023-02-27/file.parquet
+   - Or: dt=2023-02-27/file.parquet
+
+3. Downgrade to PyArrow 16:
+   pip install pyarrow==16.1.0
```

Comment:
No reason to pin it.
```suggestion
   pip install pyarrow~=16.0
```